pkg/provisioner/crdprovisioner.go

```diff
@@ -313,12 +313,12 @@ func (c *CrdProvisioner) PublishVolume(
 		if err != nil {
 			return nil, err
 		}
-		return updated.Status.PublishContext, nil
+	} else if  !errors.IsNotFound(err) {
+		return nil, err
 	} else {
-
 		azVolumeAttachment := &v1alpha1.AzVolumeAttachment{
 			ObjectMeta: metav1.ObjectMeta{
-				Name: attachmentName,
+				Name: strings.ToLower(attachmentName),
```

Comment:
`strings.ToLower` is not necessary here because `azureutils.GetAzVolumeAttachmentName` returns a lower-cased name.